File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        unordered_map<int , int> mp;
        vector<int> sortedNums = nums;
        sort(sortedNums.begin(), sortedNums.end());
        for(int i = nums.size()-1 ; i >= 0 ; i--){
            mp[sortedNums[i]] = i;
        }
        for(int i = 0 ; i < nums.size() ; i++){
            nums[i] = mp[nums[i]];
        }
        return nums;
    }
};

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: list[int]) -> list[int]:
        # Count frequency of each number
        count = [0] * 102
        for num in nums:
            count[num] += 1
            
        # Cumulative sum to find how many numbers are smaller
        for i in range(1, 102):
            count[i] += count[i - 1]
            
        # Build the result array
        return [0 if num == 0 else count[num - 1] for num in nums]
